solarwinds/stackstate_checks/solarwinds/solarwinds.py

```diff
@@ -391,11 +391,20 @@ def __init__(self, node_id, caption, ip_address, node_status, domain, layer, nod
         self.identifiers = []
         if caption == ip_address:
             # Nodes with just an IP address in the name never have a domain part
-            self._node_identifier = Identifiers.create_host_identifier(self.caption)
+            if self.component_type == "host":
+                self._node_identifier = Identifiers.create_host_identifier(self.caption)
+            else:
+                self._node_identifier = Identifiers.create_custom_identifier('urn:network_host:', self.caption)
             self.identifiers.append(self._node_identifier)
         else:
-            self._node_identifier_no_caps = Identifiers.create_host_identifier(self.caption.lower())
-            self._node_identifier_all_caps = Identifiers.create_host_identifier(self.caption.upper())
+            if self.component_type == 'host':
+                self._node_identifier_no_caps = Identifiers.create_host_identifier(self.caption.lower())
+                self._node_identifier_all_caps = Identifiers.create_host_identifier(self.caption.upper())
+            else:
+                self._node_identifier_no_caps = Identifiers.create_custom_identifier('network_host',
```

Comment:
`create_custom_identifer` here won't create `urn:network_host`. you have to pass first argument as `urn:network_host` here as well. I am not sure how did tests pass?

Reply:
It will ```    def create_custom_identifier(identifier_namespace, identifier):
        """
        creates a urn based identifier that can be used in StackState.
        args: `identifier_namespace`, `identifier`
        `identifier_namespace` is the namespace that this identifier falls in, typically this in the source or the
        component / relation type eg. kubernetes, stackstate-agent
        `identifier` is the actual identifying part of the identifier. This can also include sub-namespaces as described
        in the class definition.
        """
        return "urn:%s:/%s" % (identifier_namespace, to_string(identifier))
        ```

Reply:
@hruhek  Yeah but line 397.

Reply:
@vishalxebia Fixed